ClosedXML_Tests/Excel/NamedRanges/NamedRangesTests.cs

```diff
@@ -323,5 +323,200 @@ public void CopyNamedRangeDifferentWorksheets()
             Assert.AreEqual("Sheet2!B2:E6", copy.Ranges.First().RangeAddress.ToString(XLReferenceStyle.A1, true));
             Assert.AreEqual("Sheet2!D1:E2", copy.Ranges.Last().RangeAddress.ToString(XLReferenceStyle.A1, true));
         }
+
+        [Test]
+        public void NamedRangesBecomeInvalidOnWorksheetDeleting()
+        {
+            using (var wb = new XLWorkbook())
+            {
+                var ws1 = wb.Worksheets.Add("Sheet 1");
+                var ws2 = wb.Worksheets.Add("Sheet 2");
+                var ws3 = wb.Worksheets.Add("Sheet'3");
+
+                ws1.Range("A1:D1").AddToNamed("Named range 1", XLScope.Worksheet);
+                ws1.Range("A2:D2").AddToNamed("Named range 2", XLScope.Workbook);
+                ws2.Range("A3:D3").AddToNamed("Named range 3", XLScope.Worksheet);
+                ws2.Range("A4:D4").AddToNamed("Named range 4", XLScope.Workbook);
+                wb.NamedRanges.Add("Named range 5", new XLRanges
+                {
+                    ws1.Range("A5:D5"),
+                    ws3.Range("A5:D5")
+                });
+
+                ws2.Delete();
+                ws3.Delete();
+
+                Assert.AreEqual(1, ws1.NamedRanges.Count());
+                Assert.AreEqual("Named range 1", ws1.NamedRanges.First().Name);
+                Assert.AreEqual(XLNamedRangeScope.Worksheet, ws1.NamedRanges.First().Scope);
+                Assert.AreEqual("'Sheet 1'!$A$1:$D$1", ws1.NamedRanges.First().RefersTo);
+                Assert.AreEqual("'Sheet 1'!A1:D1", ws1.NamedRanges.First().Ranges.Single().RangeAddress.ToString(XLReferenceStyle.A1, true));
+
+                Assert.AreEqual(3, wb.NamedRanges.Count());
+
+                Assert.AreEqual("Named range 2", wb.NamedRanges.ElementAt(0).Name);
+                Assert.AreEqual(XLNamedRangeScope.Workbook, wb.NamedRanges.ElementAt(0).Scope);
+                Assert.AreEqual("'Sheet 1'!$A$2:$D$2", wb.NamedRanges.ElementAt(0).RefersTo);
+                Assert.AreEqual("'Sheet 1'!A2:D2", wb.NamedRanges.ElementAt(0).Ranges.Single().RangeAddress.ToString(XLReferenceStyle.A1, true));
+
+                Assert.AreEqual("Named range 4", wb.NamedRanges.ElementAt(1).Name);
+                Assert.AreEqual(XLNamedRangeScope.Workbook, wb.NamedRanges.ElementAt(1).Scope);
+                Assert.AreEqual("#REF!", wb.NamedRanges.ElementAt(1).RefersTo);
```

Comment:
Ah, this is wrong. Probably, when I searched the proper behavior I look into the wrong column: `Value` instead of `Refers to`:
![image](https://user-images.githubusercontent.com/19576939/40639819-179a66b2-6324-11e8-917f-9bef99a00bd7.png)